authors/apps/authentication/views.py

```diff
@@ -143,3 +146,54 @@ def update(self, request, *args, **kwargs):
         serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class PasswordResetView(CreateAPIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = EmailSerializer
+
+    def post(self, request):
+        recipient = request.data.get('email', {})
+        if not recipient:
+            return Response({"message": "Email field cannot be blank"}, status=status.HTTP_400_BAD_REQUEST)
+
+        token = jwt.encode({"email": recipient},
+                           settings.SECRET_KEY, algorithm='HS256')
+        
+        is_user_exising = User.objects.filter(email=recipient).exists()
+        if is_user_exising:
+            result = MailSender.send_email_message(recipient, token, request)
+            return Response(result, status=status.HTTP_200_OK)
+        
+        else:
+            result = {
+                'message': 'A user with the given email was not found'
+            }
+            return Response(result, status=status.HTTP_404_NOT_FOUND)
+
+class PasswordUpdateView(UpdateAPIView):
```

Comment:
Can't a `patch` in the `UserRetrieveUpdateAPIView` since a password update is just a partial update to a user's information, it doesn't require an entire view to itself.

Reply:
Yes, it's possible. But how will I handle the serializer because in the `userRetrieveUpdateAPIView`, the person must be authenticated to access the SAFE methods. While password updating should give permission to anyone who creates the request with a valid email.